docs/9_0/components/photocam.md

```diff
@@ -73,4 +75,65 @@ public class PhotoCamBean {
+`InputDeviceInfo` objects which can provide the required `deviceId` hash. The code below shows how to extract this information to populate a HTML select dropdown:
+
+```javascript
+    window.setTimeout(function() {
+	    var photoCam = PF('photoCam');
+	    var devices = photoCam.devices;
+	    if (devices) {
+	        var deviceSelector = document.querySelector("select");
+	        devices.forEach(device => {
+	            if (device.deviceId) {
+	                var option = document.createElement("option");
+	                if (device.label) {
+	                    option.text = device.label;
+	                } else {
+	                    option.text = "unamed device";
+	                }
+	                option.value = device.deviceId;
+	                deviceSelector.appendChild(option);
+	            }
+	        });
+	    } else {
+	        console.log("no devices found");
+	    }
+	}, 500);
+
+```
+
+It turns out that even browsers in regular or fast PCs may be a bit lazy to fill up the `InputDeviceInfo` collection data into the `devices` widget attribute.
+Thus, it is a good idea to wait some mills before consume these objects, for example, using the `window.setTimeout` idiom as shown above.
+
+## Error handling
+
+The default error handling is a browser native alert() popup prompting the user about some low level / browser / device undesired event.
+Lucky, `onCameraError` fallback can be defined to indicate a custom handler to deal with these errors. For instance:
+
+```javascript
+    function myCustomErrorHandler(errorObj) {
+        console.log("error", errorObj);
+    }
+```
+```xhtml
+    <p:photoCam widgetVar="photoCam" onCameraError="myCustomErrorHandler(errorObj);"/>
+```
+will swept to console any error thrown by the PhotoCam underlying engine.
+
+## Browser Compatibility 
+
+This component is strongly dependent of browser / operational system faculties and constraints. Some of known restrictions are listed below:
+
+- Navigators like Google Chrome (version 47 and later) require secure origins (HTTPS or LOCALHOST) to activate media devices functionality. Thus, unless you are accessing the page from LOCALHOST, PhotoCam is likely to do not work in a plain HTTP call. More details can be found here: https://developers.google.com/web/updates/2015/10/chrome-47-webrtc
+- Third-party browsers on iOS devices (Iphone, Ipad, etc) do not have same access to media devices as the Apple native browser Safari. Thus, on an iOS device, PhotoCam should works fine on Safari but not on Chrome or Firefox (or other no-Safari navigators). See http://www.openradar.me/33571214 for any updates on this. Note that this restriction is applied only for iOS devices. In general, third-party browsers running on OSx systems (mackbooks, iMacs and so on) don't have this same restriction.
+- PhotoCam does not support Microsoft Internet Explorer 11 or below. Other eldery browsers are likely to do not work as well.
```

Comment:
PhotoCam does not support Microsoft Internet Explorer 11 or below. Other legacy browsers are also not likely to be functional.

Reply:
Done!

Reply:
Just this one last wording change above...

Reply:
Ops! Fixed in last commit.